src/workerd/io/worker.c++

```diff
@@ -2326,8 +2214,236 @@ public:
     // delay signaling the outgoing loop until this call?
   }
 
+  void pauseIncomingMessages();
+  // Stops the delivery of CDP messages on the I/O worker thread. Called when the isolate hits a
+  // breakpoint or debugger statement.
+
+  void resumeIncomingMessages();
+  // Resumes delivery of CDP messages on the I/O worker thread. Called when execution resumes after
+  // a breakpoint or debugger statement.
+
+  bool dispatchOneMessageDuringPause();
+  // Dispatches one message whilst automatic CDP messages on the I/O worker thread is paused, called
+  // on the thread executing the isolate whilst execution is suspended due to a breakpoint or
+  // debugger statement.
+
 private:
-  kj::WebSocket& webSocket;
+  class WebSocketIoWorker final {
```

Comment:
The use of the word "Worker" in this name is a bit confusing given our product name. :)

When I saw references to "I/O worker thread" in comments above I initially assumed this meant "the thread running the worker".

Maybe "Handler" instead of "Worker"? Or some other generic word, just not "Worker".

Reply:
Thanks, done.